tests/test_config.py

```diff
@@ -1,124 +1,182 @@
-import json
 import os
 import sys
+import tempfile
+from copy import deepcopy
+from pathlib import Path
+from typing import Any
 
 import pytest
 import yaml
-from click.testing import CliRunner
+from click import ClickException
 from mock import patch
 
-from ggshield.config import Cache, Config, replace_in_keys
+from ggshield.config import (
+    AccountConfig,
+    Config,
+    InstanceConfig,
+    get_auth_config_filepath,
+    replace_in_keys,
+)
+from ggshield.utils import dashboard_to_api_url
+
+
+@pytest.fixture
+def tmp_dir_name():
```

Comment:
pytest already provides tmp_dir and tmp_path (which is even nicer) fixtures. Maybe we can use tmp_path instead?